lib/ace/edit_session.js

```diff
@@ -1135,6 +1135,11 @@ EditSession.$uid = 0;
      *
      **/
     this.insert = function(position, text) {
+        if (text == this.$bidiHandler.LRE || text == this.$bidiHandler.RLE || this.$bidiHandler.isRtlLine()) {
```

Comment:
it is better to use eventListener, instead of modifying session insert and remove, either the change event, or add a new event to https://github.com/ajaxorg/ace/blob/master/lib/ace/editor.js#L966.